src/ui/components/Timeline/Tooltip.tsx

```diff
@@ -1,37 +1,38 @@
 import React from "react";
-import { connect, ConnectedProps } from "react-redux";
-import { selectors } from "ui/reducers";
-import { UIState } from "ui/state";
+import { useSelector } from "react-redux";
+import { getIsFocusing, getNonLoadingTimeRanges } from "ui/reducers/app";
+import { getHoverTime, getZoomRegion } from "ui/reducers/timeline";
 import { getVisiblePosition } from "ui/utils/timeline";
 
-export function Tooltip({ hoverTime, zoomRegion, timelineWidth, isFocusing }: Props) {
+const getTimestamp = (time?: number) => {
+  const date = new Date(time || 0);
+  const seconds = date.getSeconds().toString().padStart(2, "0");
+  const minutes = date.getMinutes();
+
+  return `${minutes}:${seconds}`;
+};
+
+export default function Tooltip({ timelineWidth }: { timelineWidth: number }) {
+  const hoverTime = useSelector(getHoverTime);
+  const zoomRegion = useSelector(getZoomRegion);
+  const isFocusing = useSelector(getIsFocusing);
+  const nonLoadingRegion = useSelector(getNonLoadingTimeRanges);
   const shouldHideTooltip = !hoverTime || isFocusing;
+
   if (shouldHideTooltip) {
     return null;
   }
 
   let offset = getVisiblePosition({ time: hoverTime, zoom: zoomRegion }) * timelineWidth;
 
-  const time = new Date(hoverTime || 0);
-  const seconds = time.getSeconds().toString().padStart(2, "0");
-  const minutes = time.getMinutes();
+  const isHoveredOnNonLoadingRegion = nonLoadingRegion.some(
+    r => r.start <= hoverTime && r.end >= hoverTime
+  );
```

Comment:
nit: We have a new helper for this if you wanted to use it:
https://github.com/RecordReplay/devtools/blob/d7683b635fd8ea15717fa0c1e130e91c0b9e2d4c/src/ui/utils/timeline.ts#L149-L151